### PR TITLE
Remove oc client and add helper functions

### LINKED ISSUE #173: Create local_client_url() cluster method to return the local client url
The [conversation](https://project-codeflare.slack.com/archives/C04LK0G441F/p1687455584633079?thread_ts=1687449139.384029&cid=C04LK0G441F).
Basically we need need to get rid of:
```
with oc.project(cluster.config.namespace):
    routes=oc.selector("route").objects()
    rayclient_url=""
    for r in routes:
        if "rayclient" in r.name():
            rayclient_url=r.model.spec.host
ray.init(address=f"ray://{rayclient_url}", logging_level="DEBUG")
```
and create a `local_client_url()` method so we only need to do:
```
ray.init(address=cluster.local_client_url(), logging_level="DEBUG")
```

### PR DESCRIPTION
**Pull request by tedhtchang**
closes #173 
How to test:
```
# clone the branch
git clone --single-branch --branch issue173 https://github.com/tedhtchang/codeflare-sdk.git
cd codeflare-sdk/

# create python virtual env with [pyenv](https://github.com/pyenv/pyenv#automatic-installer)
pyenv install 3.8.13
pyenv virtualenv 3.8.13 localinter
pyenv local localinter

# Install python packages
pip install -r requirements.txt
pip install -r requirements-dev.txt
pip install jupyterlab
pip install -e .

# Run a sameple noteook and then Run -> Run all cells
jupyter lab demo-notebooks/interactive/local_interactive.ipynb

# clean up
pyenv uninstall localinter
# delete the cloned repo
```
Expect output similar:
image

### CODE REVIEW

**Comment by Bobbins228** [comment]
LGTM Just tested it out working as expected nice work Ted

**Review by Bobbins228 — APPROVED** [review]
LGTM Just tested it out working as expected nice work Ted

File: src/codeflare_sdk/cluster/cluster.py [hunk 1]
```diff
@@ -34,6 +34,7 @@
     RayCluster,
     RayClusterStatus,
 )
+from kubernetes import client, config
 
 from kubernetes import client, config
```
Comment (Maxusmusti) [inline src/codeflare_sdk/cluster/cluster.py:39]: Double import?

File: tests/unit_test.py [hunk 2]
```diff
@@ -362,7 +362,7 @@ def test_cluster_uris(mocker):
     mocker.patch("kubernetes.config.load_kube_config", return_value="ignore")
     mocker.patch(
         "kubernetes.client.CustomObjectsApi.list_namespaced_custom_object",
-        side_effect=uri_retreival,
+        side_effect=d,
```
Comment (Maxusmusti) [inline tests/unit_test.py:365]: what is `d` here?

**Review by Maxusmusti — CHANGES_REQUESTED** [review]
Lookg good, just needs a rebase and a couple of very small nits

**Comment by carsonmh** [comment]
Tested it out and it works as expected. LGTM!

**Review by carsonmh — APPROVED** [review]

**Pull request merged**